xls/dslx/virtualizable_file_system.h

```diff
@@ -0,0 +1,95 @@
+#ifndef XLS_DSLX_VIRTUALIZABLE_FILE_SYSTEM_H_
+#define XLS_DSLX_VIRTUALIZABLE_FILE_SYSTEM_H_
+
+#include <filesystem>  // NOLINT
+
```

Comment:
Some `#include`s missing for `<string>`, `<string_view>`, `<optional>`, StrFormat...

see clang-tidy output.